### PR TITLE
Fix.

### PR DESCRIPTION
**Pull request by hema-1046**

### CODE REVIEW

**Comment by ecki** [comment]
Can you make the commit message a bit more verbose?

File: features/core/src/main/java/org/apache/karaf/features/LocationPattern.java [hunk 1]
```diff
@@ -220,6 +220,30 @@ public boolean matches(String otherUri) {
         return match;
     }
 
+    /**
+     * Returns <code>true</code> if this location pattern strictly matches other pattern.
+     * <p>
+     * In a strict match, if this instance doesn't contain a classifier, it won't match against an otherUri which contains one.
```
Comment (ecki) [inline features/core/src/main/java/org/apache/karaf/features/LocationPattern.java]: Line length

File: features/core/src/main/java/org/apache/karaf/features/LocationPattern.java [hunk 2]
```diff
@@ -220,6 +220,30 @@ public boolean matches(String otherUri) {
         return match;
     }
 
+    /**
+     * Returns <code>true</code> if this location pattern strictly matches other pattern.
```
Comment (ecki) [inline features/core/src/main/java/org/apache/karaf/features/LocationPattern.java]: @{code true}

File: features/core/src/main/java/org/apache/karaf/features/LocationPattern.java [hunk 3]
```diff
@@ -220,6 +220,30 @@ public boolean matches(String otherUri) {
...
+     * <p>
+     * In a strict match, if this instance doesn't contain a classifier, it won't match against an otherUri which contains one.
+     *
+     * @param otherUri
```
Comment (ecki) [inline features/core/src/main/java/org/apache/karaf/features/LocationPattern.java]: If you don’t use the parameters/returntypes (which are probably needed for a public api method) then i would remove them and not keep them empty.

File: features/core/src/main/java/org/apache/karaf/features/internal/service/FeaturesProcessorImpl.java [hunk 4]
```diff
@@ -25,8 +25,10 @@
 import java.net.URL;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 import java.util.Properties;
 import java.util.Set;
```
Comment (ecki) [inline features/core/src/main/java/org/apache/karaf/features/internal/service/FeaturesProcessorImpl.java]: Is that used?

**Comment by hema-1046** [comment]
Created karaf issue [KARAF-6410] and PR.
@ecki: your comments haven't been ignored.

**Pull request closed without merging**